### PR TITLE
update: translate `useDebugValue`

### LINKED ISSUE #399: `useDebugValue`
Page: https://id.react.dev/reference/react/useDebugValue

Assignee: @arganaphang

### PR DESCRIPTION
**Pull request by arganaphang**
Closes #399 

- update: translate `hooks:useDebugValue` page

### CODE REVIEW

File: src/content/reference/react/useDebugValue.md [hunk 1]
```diff
@@ -16,11 +16,11 @@ useDebugValue(value, format?)
 
 ---
 
-## Reference {/*reference*/}
+## Referensi {/*reference*/}
 
 ### `useDebugValue(value, format?)` {/*usedebugvalue*/}
 
-Call `useDebugValue` at the top level of your [custom Hook](/learn/reusing-logic-with-custom-hooks) to display a readable debug value:
+Panggil `useDebugValue` di bagian atas [kustom Hook](/learn/reusing-logic-with-custom-hooks) Anda untuk manampilkan nilai debug yang dapat dibaca:
```
Comment (r17x) [inline src/content/reference/react/useDebugValue.md:23]: add `*` for `*debug*`

Comment (arganaphang) [reply src/content/reference/react/useDebugValue.md]: updated 👍

File: src/content/reference/react/useDebugValue.md [hunk 2]
```diff
@@ -59,11 +59,11 @@ function useOnlineStatus() {
 }
 ```
 
-This gives components calling `useOnlineStatus` a label like `OnlineStatus: "Online"` when you inspect them:
+Ini membiarkan komponen memanggil `useOnlineStatus` sebuah label seperti `OnlineStatus: "Online"` ketika Anda memeriksa mereka:
```
Comment (mazipan) [inline src/content/reference/react/useDebugValue.md:62]: Paragraf ini masih susah ta tangkap maksudnya.

File: src/content/reference/react/useDebugValue.md [hunk 3]
```diff
@@ -103,20 +103,20 @@ function subscribe(callback) {
 
 <Note>
 
-Don't add debug values to every custom Hook. It's most valuable for custom Hooks that are part of shared libraries and that have a complex internal data structure that's difficult to inspect.
+Jangan menambahkan nilai *debug* untuk setiap kustom Hook. Itu paling berharga untuk kustom Hooks yang merupakan bagian dari pustaka bersama dan memiliki struktur data internal kompleks yang sulit untuk diperiksa.
```
Comment (mazipan) [inline src/content/reference/react/useDebugValue.md:106]:
```suggestion
Jangan menambahkan nilai *debug* untuk setiap kustom Hook. Itu paling berharga untuk kustom Hooks yang merupakan bagian dari pustaka bersama dan memiliki struktur data internal yang kompleks sehingga sulit untuk diperiksa.
```

File: src/content/reference/react/useDebugValue.md [hunk 4]
```diff
@@ -103,20 +103,20 @@ function subscribe(callback) {
...
 
-You can also pass a formatting function as the second argument to `useDebugValue`:
+Anda juga bisa meneruskan fungsi pemformatan sebagai argumen kedua ke `useDebugValue`:
 
 ```js [[1, 1, "date", 18], [2, 1, "date.toDateString()"]]
 useDebugValue(date, date => date.toDateString());
 ```
 
-Your formatting function will receive the <CodeStep step={1}>debug value</CodeStep> as a parameter and should return a <CodeStep step={2}>formatted display value</CodeStep>. When your component is inspected, React DevTools will call this function and display its result.
+Fungsi pemformatan Anda akan menerima <CodeStep step={1}>nilai debug</CodeStep> sebagai sebuah parameter dan akan kembali sebuah <CodeStep step={2}>nilai tampilan yang diformat</CodeStep>. Ketika komponen Anda diperiksa, React DevTools akan memanggil fungsi ini dan menampilkan hasilnya.
```
Comment (mazipan) [inline src/content/reference/react/useDebugValue.md:120]:
```suggestion
Fungsi pemformatan Anda akan menerima <CodeStep step={1}>nilai debug</CodeStep> sebagai sebuah parameter dan akan mengembalikan sebuah <CodeStep step={2}>nilai tampilan yang telah diformat</CodeStep>. Ketika komponen Anda diperiksa, React DevTools akan memanggil fungsi ini dan menampilkan hasilnya.
```

File: src/content/reference/react/useDebugValue.md [hunk 5]
```diff
@@ -103,20 +103,20 @@ function subscribe(callback) {
...
 
-This lets you avoid running potentially expensive formatting logic unless the component is actually inspected. For example, if `date` is a Date value, this avoids calling `toDateString()` on it for every render.
+Ini memungkinkan Anda menghindari menjalankan logika pemformatan yang berpotensi mahal kecuali komponen benar-benar diperiksa. Sebagai contoh, jika `date` merupakan sebuah nilai Tanggal, ini menghindari pemanggilan `toDateString()` pada setiap render komponen.
```
Comment (mazipan) [inline src/content/reference/react/useDebugValue.md:122]: Expensive di sini kayanya lebih relevan dengan arti "berat"

File: src/content/reference/react/useDebugValue.md (hunk 2 above)
Comment (arganaphang) [reply src/content/reference/react/useDebugValue.md]: updated ya mas, boleh diperiksa lagi apakah sudah cocok, terutama bagian ini.
btw terima kasih feedback nya 👍

File: src/content/reference/react/useDebugValue.md [hunk 6]
```diff
@@ -103,20 +103,20 @@ function subscribe(callback) {
...
 
 ```js [[1, 1, "date", 18], [2, 1, "date.toDateString()"]]
 useDebugValue(date, date => date.toDateString());
 ```
 
-Your formatting function will receive the <CodeStep step={1}>debug value</CodeStep> as a parameter and should return a <CodeStep step={2}>formatted display value</CodeStep>. When your component is inspected, React DevTools will call this function and display its result.
+Fungsi pemformatan Anda akan menerima <CodeStep step={1}>nilai debug</CodeStep> sebagai sebuah parameter dan akan mengembalikan sebuah <CodeStep step={2}>nilai tampilan yang telah diformat</CodeStep>. Ketika komponen Anda diperiksa, React DevTools akan memanggil fungsi ini dan menampilkan hasilnya.
 
-This lets you avoid running potentially expensive formatting logic unless the component is actually inspected. For example, if `date` is a Date value, this avoids calling `toDateString()` on it for every render.
+Ini memungkinkan Anda menghindari menjalankan logika pemformatan yang berpotensi berat kecuali komponen benar-benar diperiksa. Sebagai contoh, jika `date` merupakan sebuah nilai Tanggal, ini menghindari pemanggilan `toDateString()` pada setiap render komponen.
```
[the remark this answers is not in the document]
Comment (arganaphang) [reply src/content/reference/react/useDebugValue.md:122]: i think `mahal` is better than `berat` in this case. How about you sir @mazipan ?

[the remark this answers is not in the document]
Comment (mhaidarhanif) [reply src/content/reference/react/useDebugValue.md]: @resir014 can also suggest, we need to clarify this later too in the glossary.

**Review by r17x — APPROVED** [review]

**Comment by r17x** [comment]
I think it can be merged but need confirm @mhaidarhanif

**Review by mazipan — APPROVED** [review]

**Comment by arganaphang** [comment]
can we merge this PR sir @resir014?

File: src/content/reference/react/useDebugValue.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (mhaidarhanif) [reply src/content/reference/react/useDebugValue.md]: @mazipan @arganaphang Oke bisa juga seperti itu, `operasi yang mahal/berat (*expensive operation*)`

cc @resir014 @r17x @zainfathoni

File: src/content/reference/react/useDebugValue.md [hunk 7]
```diff
@@ -103,20 +103,20 @@ function subscribe(callback) {
...
 
 ```js [[1, 1, "date", 18], [2, 1, "date.toDateString()"]]
 useDebugValue(date, date => date.toDateString());
 ```
 
-Your formatting function will receive the <CodeStep step={1}>debug value</CodeStep> as a parameter and should return a <CodeStep step={2}>formatted display value</CodeStep>. When your component is inspected, React DevTools will call this function and display its result.
+Fungsi pemformatan Anda akan menerima <CodeStep step={1}>nilai debug</CodeStep> sebagai sebuah parameter dan akan mengembalikan sebuah <CodeStep step={2}>nilai tampilan yang telah diformat</CodeStep>. Ketika komponen Anda diperiksa, React DevTools akan memanggil fungsi ini dan menampilkan hasilnya.
 
-This lets you avoid running potentially expensive formatting logic unless the component is actually inspected. For example, if `date` is a Date value, this avoids calling `toDateString()` on it for every render.
+Ini memungkinkan Anda menghindari menjalankan logika pemformatan yang berpotensi mahal kecuali komponen benar-benar diperiksa. Sebagai contoh, jika `date` merupakan sebuah nilai *Date*, ini menghindari pemanggilan `toDateString()` pada setiap render.
```
Comment (mhaidarhanif) [inline src/content/reference/react/useDebugValue.md:122]: To make it a bit easier to understand where it's coming from.

```suggestion
Ini memungkinkan Anda menghindari menjalankan logika pemformatan yang berpotensi mahal/berat (*expensive formatting logic*) kecuali komponen benar-benar diperiksa. Sebagai contoh, jika `date` merupakan sebuah nilai *Date*, ini menghindari pemanggilan `toDateString()` pada setiap render.
```

**Review by mhaidarhanif — CHANGES_REQUESTED** [review]
A small improvement.

**Review by mhaidarhanif — APPROVED** [review]

**Pull request merged**